client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -0,0 +1,448 @@
+			'woocommerce-admin'
+		);
+	};
+
+	openWcsModal() {
+		if ( window.wcsGetAppStore ) {
+			const wcsStore = window.wcsGetAppStore(
+				'wc-connect-create-shipping-label'
+			);
+			const state = wcsStore.getState();
+			const { orderId } = this.state;
+			const siteId = state.ui.selectedSiteId;
+
+			wcsStore.dispatch( {
+				type: 'WOOCOMMERCE_SERVICES_SHIPPING_LABEL_OPEN_PRINTING_FLOW',
+				orderId,
+				siteId,
+			} );
+			wcsStore.dispatch( {
+				type: 'NOTICE_CREATE',
+				notice: {
+					duration: 10000,
+					status: 'is-success',
+					text: __(
+						'Plugin installed and activated',
+						'woocommerce-admin'
+					),
+				},
+			} );
+			document.getElementById(
+				'woocommerce-admin-print-label'
+			).style.display = 'none';
+		}
+	}
+
+	render() {
+		const { isDismissModalOpen, showShippingBanner } = this.state;
+		if ( ! showShippingBanner ) {
+			return null;
+		}
+
+		return (
+			<div>
+				<div className="wc-admin-shipping-banner-container">
+					<img
+						className="wc-admin-shipping-banner-illustration"
+						src={ wcAdminAssetUrl + 'shippingillustration.svg' }
+						alt={ __( 'Shipping ', 'woocommerce-admin' ) }
+					/>
+					<Button
+						disabled={ this.state.isShippingLabelButtonBusy }
+						isPrimary
+						isBusy={ this.state.isShippingLabelButtonBusy }
+						onClick={ this.createShippingLabelClicked }
+					>
+						{ __( 'Create shipping label' ) }
+					</Button>
+					<h3>
+						{ __(
+							'Print discounted shipping labels with a click.',
+							'woocommerce-admin'
+						) }
+					</h3>
+					<p>
+						{ interpolateComponents( {
+							mixedString: this.getInstallText(),
+							components: {
+								tosLink: (
+									<ExternalLink
+										href="https://wordpress.com/tos"
+										target="_blank"
+										type="external"
+									/>
+								),
+								wcsLink: (
+									<ExternalLink
+										href="https://woocommerce.com/products/shipping/"
+										target="_blank"
+										type="external"
+										onClick={
+											this.woocommerceServiceLinkClicked
+										}
+									/>
+								),
+							},
+						} ) }
+					</p>
+					<SetupNotice
+						isSetupError={ this.isSetupError() }
+						errorReason={ this.setupErrorReason() }
+					/>
+					<button
+						onClick={ this.openDismissModal }
+						type="button"
+						className="notice-dismiss"
+						disabled={ this.state.isShippingLabelButtonBusy }
+					>
+						<span className="screen-reader-text">
+							{ __(
+								'Close Print Label Banner.',
+								'woocommerce-admin'
+							) }
+						</span>
+					</button>
+				</div>
+				<DismissModal
+					visible={ isDismissModalOpen }
+					onClose={ this.closeDismissModal }
+					onCloseAll={ this.hideBanner }
+					trackElementClicked={ this.trackElementClicked }
+				/>
+			</div>
+		);
+	}
+}
+
```

Comment:
Would be nice to have a `propTypes` declaration somewhere for this component. Guessing it would help enforce a few props as required that are anticipated.

Reply:
I'll add one